napalm_junos/junos.py

```diff
@@ -331,6 +331,11 @@ def get_environment(self):
         for sensor_object, object_data in environment.items():
             structured_object_data = {k: v for k, v in object_data}
 
+            if structured_object_data['class']:
```

Comment:
You can write this as:

```
current_class = structured_object_data['class'] or current_class
structured_object_data['class'] = current_class
```
Actually, you don't even need the second statement. You could just change the occurrences of `structured_object_data['class']` with `current_class` so the code would start like:

```
current_class = structured_object_data['class'] or current_class
if current_class == 'Power':
...
```